docs/plugin-architecture.md

```diff
@@ -0,0 +1,307 @@
+# Plugin Architecture Documentation
+
+This document provides a comprehensive overview of the WooCommerce PayPal Payments plugin architecture, explaining its modular design and how the various components work together.
+
+## Overview
+
+The WooCommerce PayPal Payments plugin is built using a modular architecture powered by the [Syde Modularity](https://github.com/inpsyde/modularity) framework. This design provides:
+
+- **Modular Structure**: Each feature is contained within its own module with clear boundaries
+- **Dependency Injection**: PSR-11 container for service management and dependency resolution
+- **Feature Flags**: Dynamic module loading based on environment variables and filters
+- **Extensibility**: Well-defined extension points for customization and enhancement
+- **Maintainability**: Clear separation of concerns and consistent patterns
+
+## Core Components
+
+### Main Plugin File
+
+The plugin initialization begins in `woocommerce-paypal-payments.php`, which:
+- Loads the Composer autoloader
+- Checks for class existence to prevent conflicts
+- Contains plugin metadata and constants definitions
+
+### Bootstrap System
+
+The bootstrap process is handled by `bootstrap.php`, which:
+
+```php
+return function (
+    string $root_dir,
+    array $additional_containers = array(),
+    array $additional_modules = array()
+): ContainerInterface {
+    // Load modules from modules.php
+    $modules = ( require "$root_dir/modules.php" )( $root_dir );
+    
+    // Apply filters for customization
+    $modules = apply_filters( 'woocommerce_paypal_payments_modules', $modules );
+    
+    // Initialize plugin with Inpsyde Modularity
+    $properties = PluginProperties::new( "$root_dir/woocommerce-paypal-payments.php" );
+    $bootstrap = Package::new( $properties );
+    
+    foreach ( $modules as $module ) {
+        $bootstrap->addModule( $module );
+    }
+    
+    $bootstrap->boot();
+    return $bootstrap->container();
+};
+```
+
+### PPCP Container
+
+The global `PPCP` class (`src/PPCP.php`) provides access to the dependency injection container:
+
+```php
+class PPCP {
+    private static $container = null;
+    
+    public static function container(): ContainerInterface {
+        if ( ! self::$container ) {
+            throw new LogicException( 'No PPCP container, probably called too early when the plugin is not initialized yet.' );
+        }
+        return self::$container;
+    }
+}
+```
+
+This allows modules to access services via `PPCP::container()->get('service.id')` after initialization.
```

Comment:
Modules already have the container (and usually pass it where needed). `PPCP::` is more about some third-party interop, such as `api/order-functions.php` or some tests.